dstep/driver/Application.d

```diff
@@ -26,280 +26,163 @@ import clang.Index;
     }
 
-    protected override void run ()
+    public this ( AppConfig config )
     {
-        handleArguments;
+        this.config = config;
+    }
 
-        if (arguments.argument.input.hasValue)
+    public void run ()
+    {
+        if (config.input_files.length <= 1)
         {
-            auto inputFiles = arguments.argument.input.values[1 .. $];
-            string outputDir = null;
-            if (inputFiles.length > 0)
-            {
-                outputDir = arguments.output;
-                if (arguments.output.any() && !exists(outputDir))
-                    mkdirRecurse(outputDir);
-            }
+            string outputFilename;
+            auto fileName = config.input_files.length ? config.input_files[0] : "";
 
-            foreach(string fileName; inputFiles)
-            {
-                auto conversionTask = task!startParsingFile(language, argsToRestore, fileName, true,
-                                                            arguments, compiler, compilerArgs, outputDir);
-                conversionTask.executeInNewThread();
-            }
+            if (config.output.length)
+                outputFilename = config.output;
+            else
+                outputFilename = defaultOutputFilename(fileName, true);
 
-            startParsingFile(language, argsToRestore, arguments.argument.input.values[0], false, arguments,
-                             compiler, compilerArgs, outputDir);
-        }
+            string outputDir = Path.dirName(outputFilename);
+            if (!exists(outputDir))
+                mkdirRecurse(outputDir);
 
+            auto conversionTask = task!startParsingFile(config,
+                fileName, outputFilename);
+            conversionTask.executeInNewThread();
+        }
         else
-            startParsingFile(language, argsToRestore, "", false, arguments, compiler, compilerArgs, null);
-    }
-
-    protected override void setupArguments ()
-    {
-        arguments.sloppy = true;
-        arguments.passThrough = true;
-
-        arguments.argument("input", "input files").params(1, int.max);
-
-        arguments('o', "output", "Write output to")
-            .params(1)
-            .defaults("");
+        {
+            foreach(string fileName; config.input_files)
+            {
+                import std.exception : enforce;
+                enforce(config.output.length);
 
-        arguments('x', "language", "Treat subsequent input files as having type <language>")
-            .params(1)
-            .restrict("c", "c-header", "objective-c", "objective-c-header")
-            .on(&handleLanguage);
+                string outputFilename = Path.buildPath(config.output,
+                    defaultOutputFilename(fileName, false));
 
-        arguments("no-comments", "Disable translation of comments.");
+                string outputDir = Path.dirName(outputFilename);
+                if (!exists(outputDir))
+                    mkdirRecurse(outputDir);
```

Comment:
Why was this code moved here? It's also duplicated in the `if` and `else` statements.